frontend/src/component/common/NewConstraintAccordion/NewConstraintAccordionList/NewConstraintAccordionList.tsx

```diff
@@ -0,0 +1,265 @@
+                    state={state}
+                />
+                <ConditionallyRender
+                    condition={Boolean(showCreateButton && onAdd)}
+                    show={
+                        <div>
+                            <StyledAddCustomLabel>
+                                <p>Add any number of constraints</p>
+                                <StyledHelpWrapper
+                                    title='View constraints documentation'
+                                    arrow
+                                >
+                                    <a
+                                        href={
+                                            'https://docs.getunleash.io/reference/strategy-constraints'
+                                        }
+                                        target='_blank'
+                                        rel='noopener noreferrer'
+                                    >
+                                        <StyledHelp />
+                                    </a>
+                                </StyledHelpWrapper>
+                            </StyledAddCustomLabel>
+                            <Button
+                                type='button'
+                                onClick={onAdd}
+                                variant='outlined'
+                                color='primary'
+                                data-testid='ADD_CONSTRAINT_BUTTON'
+                            >
+                                Add constraint
+                            </Button>
+                        </div>
+                    }
+                />
+            </StyledContainer>
+        );
+    },
+);
+
+interface IConstraintList {
+    constraints: IConstraint[];
+    setConstraints?: React.Dispatch<React.SetStateAction<IConstraint[]>>;
+    state: IUseWeakMap<IConstraint, IConstraintAccordionListItemState>;
+}
+
+export const NewConstraintAccordionList = forwardRef<
+    IConstraintAccordionListRef | undefined,
+    IConstraintList
+>(({ constraints, setConstraints, state }, ref) => {
+    const { context } = useUnleashContext();
+
+    const onEdit =
+        setConstraints &&
+        ((constraint: IConstraint) => {
+            state.set(constraint, { editing: true });
+        });
+
+    const onRemove =
+        setConstraints &&
+        ((index: number) => {
+            const constraint = constraints[index];
+            state.set(constraint, {});
+            setConstraints(
+                produce((draft) => {
+                    draft.splice(index, 1);
+                }),
+            );
+        });
+
+    const onSave =
+        setConstraints &&
+        ((index: number, constraint: IConstraint) => {
+            state.set(constraint, {});
+            setConstraints(
+                produce((draft) => {
+                    draft[index] = constraint;
+                }),
+            );
+        });
+
+    const onAutoSave =
```

Comment:
Follow the same pattern as save, but keep `editing: true` set. For the record, I don't like the pattern we use here. But if I was to refactor this I'd add another two weeks to this project. For now I'll record it as technical debt.